src/main/java/refactoringml/db/HibernateConfig.java

```diff
@@ -1,89 +1,175 @@
 package refactoringml.db;
 
+import java.util.Optional;
+import java.util.Properties;
+
+import com.zaxxer.hikari.HikariDataSource;
+
 import org.hibernate.SessionFactory;
 import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.AvailableSettings;
 import org.hibernate.cfg.Configuration;
-import org.hibernate.cfg.Environment;
 import org.hibernate.service.ServiceRegistry;
 
-import javax.persistence.CascadeType;
-import javax.persistence.ManyToOne;
-import java.util.Properties;
-
-public class HibernateConfig {
+public abstract class HibernateConfig {
 
-	private static SessionFactory sessionFactory;
+	protected final Configuration configuration = new Configuration();
 
-	public SessionFactory getSessionFactory(String url, String user, String pwd) {
+	public static SessionFactory getSessionFactory(String url, String user, String pwd) {
```

Comment:
In the old implementation, the URL for the DB could also include additional information e.g. the port, usage of ssl for authenticiation, DB time zone etc. How do you handle it in this implementation?

Reply:
I made sure to leave the old API intact so calling the service would be possible in the original way and added functionality if wanted for connectionpools etc. The url is passed as is and the tests use some of these options you mention